mindmeld/converter/dialogflow.py

```diff
@@ -0,0 +1,375 @@
+            new_dict = {}
+            while ('value' in item) and (item['value'] in item['synonyms']):
+                item['synonyms'].remove(item['value'])
+            new_dict['whitelist'] = item['synonyms']
+            new_dict['cname'] = item['value']
+            mapping_dict["entities"].append(new_dict)
+
+            target_gazetteer.write(item['value'] + "\n")
+
+        json.dump(mapping_dict, target_mapping, ensure_ascii=False, indent=2)
+
+        source_en.close()
+        target_gazetteer.close()
+        target_mapping.close()
+
+    def _create_intents_directories(self, intents):
+        """ Creates directories + files for all languages/files."""
+
+        for languages in intents.values():
+            for language, sub in languages.items():
+                dialogflow_intent_file = os.path.join(self.dialogflow_project_directory,
+                                                      "intents", sub + ".json")
+
+                mindmeld_intent_directory_name = self.clean_check(sub, self.intents_list)
+                mindmeld_intent_directory = os.path.join(self.mindmeld_project_directory,
+                                                         "domains", "general",
+                                                         mindmeld_intent_directory_name)
+
+                self.create_directory(mindmeld_intent_directory)
+
+                self._create_intent_file(dialogflow_intent_file,
+                                         mindmeld_intent_directory,
+                                         language)
+
+    def _create_intent_file(self, dialogflow_intent_file, mindmeld_intent_directory, language):
+        source_en = open(dialogflow_intent_file, 'r')
+        target_test = open(os.path.join(mindmeld_intent_directory, "test.txt"), 'w')
+        target_train = open(os.path.join(mindmeld_intent_directory, "train.txt"), 'w')
+
+        datastore = json.load(source_en)
+        all_text = []
+
+        for usersay in datastore:
+            sentence = ""
+            for texts in usersay["data"]:
+                df_text = texts["text"]
+                if "meta" in texts and texts["meta"] != "@sys.ignore":
+                    df_meta = texts["meta"]
+
+                    if re.match("(@sys.).+", df_meta):  # if text is a dialogflow sys entity
+                        if df_meta in DialogFlowConverter.sys_entity_map:
+                            mm_meta = DialogFlowConverter.sys_entity_map[df_meta]
+                        else:
+                            mm_meta = "[DNE: {sysEntity}]".format(sysEntity=df_meta[1:])
+                            logger.info("Unfortunately mindmeld does not currently support"
+                                        "%s as a sys entity."
+                                        "Please create an entity for this.", df_meta[1:])
+
+                        intent = self.clean_name(mm_meta) + "_entries_" + language
+                        part = "{" + df_text + "|" + intent + "}"
```

Comment:
I am confused here: in `{singapore|city}`, the right side of the pipe is an entity type, not an intent, so why are we adding the intent type here? Is this by design?

Reply:
My bad, I incorrectly labeled this. I have just renamed the variable intent to entity_type. The logic was correct and intent (now entity_type) is set to the df_text's entity type.